### PR TITLE
Create logging object for logTransporter instances

### PR DESCRIPTION
**Pull request by GermanBluefox**
Fixing: https://github.com/ioBroker/ioBroker.admin/issues/1907

### CODE REVIEW

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 1]
```diff
@@ -3203,8 +3203,8 @@ export class AdapterClass extends EventEmitter {
         const { options, callback, obj } = _options;
         let { id } = _options;
 
-        obj.from = obj.from || 'system.adapter.' + this.namespace;
-        obj.user = obj.user || (options ? options.user : '') || SYSTEM_ADMIN_USER;
+        obj.from = obj.from || `system.adapter.${this.namespace}`;
+        obj.user = obj.user || (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:3207]:
```suggestion
        obj.user = obj.user || options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 2]
```diff
@@ -3370,7 +3370,7 @@ export class AdapterClass extends EventEmitter {
             }
 
             obj.from = obj.from || `system.adapter.${this.namespace}`;
-            obj.user = obj.user || (options ? options.user : '') || SYSTEM_ADMIN_USER;
+            obj.user = obj.user || (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:3373]:
```suggestion
            obj.user = obj.user || options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 3]
```diff
@@ -3379,7 +3379,7 @@ export class AdapterClass extends EventEmitter {
             return adapterObjects.setObject(id, obj, options, callback);
         } else {
             obj.from = obj.from || `system.adapter.${this.namespace}`;
-            obj.user = obj.user || (options ? options.user : '') || SYSTEM_ADMIN_USER;
+            obj.user = obj.user || (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:3382]:
```suggestion
            obj.user = obj.user || options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 4]
```diff
@@ -4893,7 +4893,7 @@ export class AdapterClass extends EventEmitter {
                 obj.from = `system.adapter.${this.namespace}`;
             }
             if (!obj.user) {
-                obj.user = (options ? options.user : '') || SYSTEM_ADMIN_USER;
+                obj.user = (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:4896]:
```suggestion
                obj.user = options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 5]
```diff
@@ -5434,8 +5434,8 @@ export class AdapterClass extends EventEmitter {
                 } else if (obj) {
                     if (!obj.common.members.includes(objId)) {
                         obj.common.members.push(objId);
-                        obj.from = 'system.adapter.' + this.namespace;
-                        obj.user = (options ? options.user : '') || SYSTEM_ADMIN_USER;
+                        obj.from = `system.adapter.${this.namespace}`;
+                        obj.user = (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:5438]:
```suggestion
                        obj.user = options?.user  || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 6]
```diff
@@ -5460,8 +5460,8 @@ export class AdapterClass extends EventEmitter {
                         // @ts-expect-error
                         obj.common.members.push(objId);
 
-                        obj.from = 'system.adapter.' + this.namespace;
-                        obj.user = (options ? options.user : '') || SYSTEM_ADMIN_USER;
+                        obj.from = `system.adapter.${this.namespace}`;
+                        obj.user = (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:5464]:
```suggestion
                        obj.user = options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 7]
```diff
@@ -5589,7 +5589,7 @@ export class AdapterClass extends EventEmitter {
                                 if (pos !== -1) {
                                     obj.common.members.splice(pos, 1);
                                     obj.from = `system.adapter.${this.namespace}`;
-                                    obj.user = (options ? options.user : '') || SYSTEM_ADMIN_USER;
+                                    obj.user = (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:5592]:
```suggestion
                                    obj.user = options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 8]
```diff
@@ -6148,8 +6148,8 @@ export class AdapterClass extends EventEmitter {
 
                 if (!obj.common.members.includes(objId)) {
                     obj.common.members.push(objId);
-                    obj.from = 'system.adapter.' + this.namespace;
-                    obj.user = (options ? options.user : '') || SYSTEM_ADMIN_USER;
+                    obj.from = `system.adapter.${this.namespace}`;
+                    obj.user = (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:6152]:
```suggestion
                    obj.user = options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 9]
```diff
@@ -6166,8 +6166,8 @@ export class AdapterClass extends EventEmitter {
                     // @ts-expect-error cast to enum object
                     if (!obj.common.members.includes(objId)) {
                         obj.common.members!.push(objId);
-                        obj.from = 'system.adapter.' + this.namespace;
-                        obj.user = (options ? options.user : '') || SYSTEM_ADMIN_USER;
+                        obj.from = `system.adapter.${this.namespace}`;
+                        obj.user = (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:6170]:
```suggestion
                        obj.user = options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 10]
```diff
@@ -6316,8 +6316,8 @@ export class AdapterClass extends EventEmitter {
                             const pos = obj.common.members.indexOf(objId);
                             if (pos !== -1) {
                                 obj.common.members.splice(pos, 1);
-                                obj.from = 'system.adapter.' + this.namespace;
-                                obj.user = (options ? options.user : '') || SYSTEM_ADMIN_USER;
+                                obj.from = `system.adapter.${this.namespace}`;
+                                obj.user = (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:6320]:
```suggestion
                                obj.user = options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 11]
```diff
@@ -7405,7 +7405,7 @@ export class AdapterClass extends EventEmitter {
             typeof stateObj.from === 'string' && stateObj.from !== ''
                 ? stateObj.from
                 : `system.adapter.${this.namespace}`;
-        stateObj.user = (options ? options.user : '') || SYSTEM_ADMIN_USER;
+        stateObj.user = (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:7408]:
```suggestion
        stateObj.user = options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 12]
```diff
@@ -8159,7 +8159,7 @@ export class AdapterClass extends EventEmitter {
         // if state.from provided, we use it else, we set default property
         state.from =
             typeof state.from === 'string' && state.from !== '' ? state.from : `system.adapter.${this.namespace}`;
-        state.user = (options ? options.user : '') || SYSTEM_ADMIN_USER;
+        state.user = (options && options.user) || SYSTEM_ADMIN_USER;
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:8162]:
```suggestion
        state.user = options?.user || SYSTEM_ADMIN_USER;
```

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 13]
```diff
@@ -8476,7 +8476,7 @@ export class AdapterClass extends EventEmitter {
...
```
Comment (foxriver76) [inline packages/adapter/src/lib/adapter/adapter.ts:8479]:
```suggestion
        state.user = options?.user || SYSTEM_ADMIN_USER;
```

**Review by foxriver76 — CHANGES_REQUESTED** [review]

**Comment by foxriver76** [comment]
Just small optimizations else LGTM

**Review by foxriver76 — APPROVED** [review]

File: packages/adapter/src/lib/adapter/adapter.ts (hunk 1 above)
Comment (GermanBluefox) [reply packages/adapter/src/lib/adapter/adapter.ts]: These are typescript tricks :)

Comment (foxriver76) [reply packages/adapter/src/lib/adapter/adapter.ts]: lol yeah I have commited it for you we can merge

**Pull request merged**